docs/testplan/platform/nvidia_new_thermal_control_algorithm_test_plan.md

```diff
@@ -0,0 +1,155 @@
+# Nvidia new hw thermal control algorithm test plan
+
+## Overview
+
+Originally, there are two major parts of SONiC thermal control:
+-   thermalctld runs thermal policies in user space to handle thermal event
+-	Kernel runs step wise thermal algorithm which monitors thermal zones(asic, module, gearbox) and manages cooling devices(fan).
+
+Currently, we will move kernel part thermal algorithm to user space and extend existing hw-mgmt thermal control service. 
+The new hw-mgmt thermal control service contains below functions:
+-	A thermal algorithm to replace kernel step wise algorithm
+-	Thermal event handler which “equals” to existing SONiC thermal policies
+-	New logic such as handling memory/CPU thermal zones, etc.
+
+This feature shall migrate the new hw-mgmt thermal control service to SONiC:
+-	Enable hw-mgmt thermal algorithm on SONiC side
+-	Remove existing SONiC thermal policies configuration from Nvidia platform (thermal_control.json)
+-	Remove thermal policies related code from Nvidia platform API
+
+### Scope
+
+The test is to verify new hw thermal control algorithm works as what as expected on nvidia platform
+
+### Scale / Performance
+
+No scale/performance test involved in this test plan
+
+### Related **DUT** CLI commands
+```
+service hw-management start/stop/status
+service hw-management-tc start/stop/status
+```
+
+
+### Supported topology
+The tests will be supported on any topo.
+
+
+### Test cases #1 - Check tc service behaviors after reboot
+1. Set hw-management-tc service to enabled state
+2. Reboot system( Randomly select cold,warm,fast, config reload)
+3. Wait for TC init (~120 sec after login prompt)
+4. Check syslog messages
+5. Check service status 
+
+### Test cases #2 - Check tc service behaviors when start and stop TC service
+1. Stop hw-management-tc service
+2. Check syslog and service state
+3. Check PWM value. It should be at maximum.
+4. Start hw-management-tc service.
+5. Immediately after start  - check PWM value . It should be at maximum
+6. wait 120 sec for svc init
+7. Check syslog and service state
+8. Check PWM value. It should be in range 20~35% (not maximum)
+
+### Test cases #3 - Check tc service behaviors when hw-management is not fully initialized.
+1. Stop hw-management-tc service
+2. Stop hw-management service
+3. Start hw-management-tc service
+4. Check PWM value. It should be at maximum
+5. Wait 3 minutes
+6. Check PWM value. It should be at maximum
+
+### Test cases #4 - Check PWM is the correct value when suspend and resume TC 
+1. Start TC svc.
+2. Check PWM in a low value like range 20~35% 
+3. Set TC suspend with echo 1 >> /var/run/hw-management/config/suspend & wait 30sec
+4. Check PWM is maximum (100%)
+5. Set TC resume with echo 0 >> /var/run/hw-management/config/suspend & wait 30sec
+6. Check PWM in a low value like range 20~35% 
+
+### Test cases #5 - Temperature sweep 
```

Comment:
Different thermal zone may have different polling interval, for example: it polls module temperature every 20 seconds, it polls CPU temperature every 3 seconds... Should we test the interval here?

Reply:
It has included it: Wait poll_time secs